tools/api-markdown-documenter/src/LintApiModel.ts

```diff
@@ -0,0 +1,308 @@
+
+	logger.verbose("API model loaded.");
+
+	logger.verbose("Linting API model...");
+
+	const result = lintApiItem(apiModel, apiModel, optionsWithDefaults);
+
+	logger.verbose("API model linting completed.");
+	return result;
+}
+
+/**
+ * Recursively validates the given API item and all its descendants within the API model.
+ */
+function lintApiItem(
+	apiItem: ApiItem,
+	apiModel: ApiModel,
+	options: Required<LintApiModelOptions>,
+): LinterResult {
+	// eslint-disable-next-line unicorn/prevent-abbreviations
+	let inheritDocReferenceError: InheritDocReferenceError | undefined;
+	const linkReferenceErrors: LinkReferenceError[] | undefined = undefined;
+
+	// If the item is documented, check its link tags for errors.
+	if (options.checkReferences && apiItem instanceof ApiDocumentedItem) {
+		// Check `{@inheritDoc}` tag
+		inheritDocReferenceError = checkInheritDocTag(apiItem, apiModel);
+
+		// TODO: Check `{@link}` tags
+	}
+
+	const myResult: LinterResult =
+		inheritDocReferenceError === undefined && linkReferenceErrors === undefined
+			? successResult
+			: {
+					success: false,
+					inheritDocReferenceErrors:
+						inheritDocReferenceError === undefined ? [] : [inheritDocReferenceError],
+					linkReferenceErrors: linkReferenceErrors ?? [],
+			  };
+
+	// If the item has children, recursively validate them.
+	let membersResult: LinterResult = successResult;
+	if (ApiItemContainerMixin.isBaseClassOf(apiItem)) {
+		const memberResults = apiItem.members.map((member) =>
+			lintApiItem(member, apiModel, options),
+		);
+
+		// eslint-disable-next-line unicorn/no-array-reduce
+		membersResult = memberResults.reduce(
+			(previous, current) => mergeLinterResults(previous, current),
+			successResult,
+		);
+	}
+
+	return mergeLinterResults(myResult, membersResult);
+}
+
+/**
+ * Merges two {@link LinterResult}s into a single result.
+ */
+function mergeLinterResults(a: LinterResult, b: LinterResult): LinterResult {
+	if (a.success && b.success) {
+		return successResult;
+	}
+
+	if (a.success) {
+		return b;
+	}
+
+	if (b.success) {
+		return a;
+	}
+
+	// eslint-disable-next-line unicorn/prevent-abbreviations
+	const inheritDocReferenceErrors = [
+		...a.inheritDocReferenceErrors,
+		...b.inheritDocReferenceErrors,
+	];
+	const linkReferenceErrors = [...a.linkReferenceErrors, ...b.linkReferenceErrors];
+
+	return {
+		success: false,
+		inheritDocReferenceErrors,
+		linkReferenceErrors,
+	};
+}
+
+/**
+ * Checks the provided API item's `{@inheritDoc}` tag, ensuring that the target reference is valid within the API model.
+ */
+// eslint-disable-next-line unicorn/prevent-abbreviations
+function checkInheritDocTag(
+	apiItem: ApiDocumentedItem,
+	apiModel: ApiModel,
+): InheritDocReferenceError | undefined {
+	// eslint-disable-next-line unicorn/prevent-abbreviations
+	const inheritDocTag: DocInheritDocTag | undefined = apiItem.tsdocComment?.inheritDocTag;
+
+	if (inheritDocTag?.declarationReference !== undefined) {
```

Comment:
Nit: I'd check for equality instead and immediately `return undefined` and let the try-catch live at the same indentation level as the `if` check.